### DISCUSSION TITLE
Install Manually

### DISCUSSION THREAD

**Discussion by gt2416** [hargata/lubelog#564] [Q&A] [answered]
Could someone give me some pointers on where to start if I want to run the webapp manually on a alpine VM. The docker image is old and using an entire windows VM to run this is too taxing on resources. 

Will downloading the wwwroot and pointing my webserver at it do the trick or is there other things that need to be done?
I plan on using sqlite DB for ease of use and caddy as my webserver.

**Comment by hargata** [comment, marked as answer]
@gt2416 no, simply downloading and pointing at the wwwroot won't work because the app is more than a static site and has an actual backend. The docker image in both GHCR and DockerHub is up to date at v1.3.5 unless you're referring to an image compiled by a third party?

Anyways, if you mean that the provided Docker image isn't compatible with the architecture of your system(armv7/x86/etc), the app is written in dotnet 8 which is cross-platform compatible so it should technically run on everything dotnet supports, but you will need to build it yourself.

Your options are as following:

- Clone the repository and build your own Docker image by modifying Dockerfile
- Clone the repository and build the app natively using `dotnet publish`

Option 1 is relatively straightforward, there are quite a few examples floating around the web.

Option 2 requires installing the dotnet SDK and running some commands, for example:

`dotnet publish -r linux-arm32`

which will build a copy of LubeLogger that will run on 32-bit Linux. I don't have a Linux machine to test this out on so a lot of this is just based on what I read on MS Docs, but I have successfully built and ran LubeLogger on MacOS so I don't see why Linux would be an issue.

**Comment by gt2416** [comment]
Thank a bunch! Yes the docker image is updated. For some reason Truenas refused to update it saying the image was already up to date. After forcing a image download I can see I am on the latest version. 
For my needs the easiest option seems like making a alpine VM with docker. Thanks !
